src/algorithms/toposort.rs

```diff
@@ -123,20 +229,35 @@ impl<'graph> FusedIterator for TopoSort<'graph> {}
 
 #[cfg(test)]
 mod test {
-    use crate::{algorithms::toposort, Direction, PortGraph};
+    use super::*;
+
+    use crate::{Direction, PortGraph};
 
     #[test]
     fn small_toposort() {
         let mut graph = PortGraph::new();
         let node_a = graph.add_node(2, 3);
         let node_b = graph.add_node(3, 2);
+        let node_c = graph.add_node(3, 2);
+        let node_d = graph.add_node(3, 2);
 
         // Add two edges between node A and B
         graph.link_nodes(node_a, 0, node_b, 0).unwrap();
         graph.link_nodes(node_a, 1, node_b, 1).unwrap();
+        graph.link_nodes(node_b, 0, node_c, 0).unwrap();
+        graph.link_nodes(node_c, 0, node_d, 0).unwrap();
```

Comment:
I think in this graph the port filter would mean both node_c and node_d are ignored? So then the node filter is not being tested.

I suggest some modification that tests both filters

Reply:
The `todosort_filtered` call had `node_d` as a source, to check it was ignored on the `TopoSort` initialization. I added an extra ignored non-source node to check that code path too.